spec/03-plugins/11-correlation-id/02-schema_spec.lua

```diff
@@ -0,0 +1,184 @@
+    assert.is_truthy(ws)
+    plugin_id = uuid.generate_v4()
+    local sql = render([[
+      INSERT INTO plugins (id, name, config, enabled, ws_id) VALUES
+        ('$(ID)', '$(PLUGIN_NAME)', $(CONFIG)::jsonb, TRUE, '$(WS_ID)');
+      COMMIT;
+    ]], {
+      ID = plugin_id,
+      PLUGIN_NAME = plugin_name,
+      CONFIG = pgmoon_json.encode_json(plugin_config),
+      WS_ID = ws.id,
+    })
+
+    local res, err = db.connector:query(sql)
+    assert.is_nil(err)
+    assert.is_not_nil(res)
+  end)
+
+  describe("in traditional mode", function()
+    lazy_setup(function()
+      assert(helpers.start_kong({
+        database = strategy,
+      }))
+    end)
+
+    before_each(function()
+      admin_client = helpers.admin_client()
+    end)
+
+    after_each(function()
+      admin_client:close()
+    end)
+
+    lazy_teardown(function()
+      if admin_client then
+        admin_client:close()
+      end
+      assert(helpers.stop_kong())
+    end)
+
+    it("auto-complete generator if it is `null` in database", function()
+      local sql = 'SELECT config FROM plugins WHERE id=\''.. plugin_id ..'\';'
+      local res, err = db.connector:query(sql)
+      assert.is_nil(err)
+      assert.is_nil(res[1].generator)
+
+      res = admin_client:get("/plugins")
+      res = cjson.decode(assert.res_status(200, res))
+      assert.equals(res.data[1].config.generator, "uuid#counter")
+    end)
+  end)
+
+  describe("in hybrid mode", function()
+    local route
+    setup(function()
+      route = bp.routes:insert({
+        hosts = {"example.com"},
+      })
+      bp.plugins:insert {
+        name    = "request-termination",
+        route   = { id = route.id },
+        config  = {
+          status_code = 200,
+        },
+      }
+      local sql = render([[
+        UPDATE plugins SET route_id='$(ROUTE_ID)', 
+        protocols=ARRAY['grpc','grpcs','http','https'], 
+        cache_key='$(CACHE_KEY)' 
+        WHERE id='$(ID)';
+        COMMIT;
+      ]], {
+        ROUTE_ID = route.id,
+        CACHE_KEY = "plugins:correlation-id:"..route.id.."::::"..ws.id,
+        ID = plugin_id,
+      })
+      local _, err = db.connector:query(sql)
+      assert.is_nil(err)
+
+      assert(helpers.start_kong({
+        role = "control_plane",
+        cluster_cert = "spec/fixtures/kong_clustering.crt",
+        cluster_cert_key = "spec/fixtures/kong_clustering.key",
+        database = strategy,
+        prefix = "servroot",
+        cluster_listen = "127.0.0.1:9005",
+        nginx_conf = "spec/fixtures/custom_nginx.template",
+      }))
+
+      assert(helpers.start_kong({
+        role = "data_plane",
+        database = "off",
+        prefix = "servroot2",
+        cluster_cert = "spec/fixtures/kong_clustering.crt",
+        cluster_cert_key = "spec/fixtures/kong_clustering.key",
+        cluster_control_plane = "127.0.0.1:9005",
+        proxy_listen = "0.0.0.0:9002",
+        status_listen = "127.0.0.1:9100",
+      }))
```

Comment:
I recommend moving `start_kong` kong to the lazy_setup phase, otherwise, kong will not normally stop if someone runs a case of this test file.

Reply:
done